ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala

```diff
@@ -77,17 +80,117 @@ class Endpoints(
   private def toRoute(res: => Future[Error \/ SearchResult[Error \/ JsValue]]): Route =
     responseToRoute(httpResponse(res))
 
+  private def mkRequestLogMsg(request: HttpRequest, remoteAddress: RemoteAddress) =
+    s"Incoming ${request.method.value} request on ${request.uri} from $remoteAddress"
+
+  private def mkResponseLogMsg(response: HttpResponse) =
+    s"Responding to client with HTTP ${response.status}"
+
+  // Always put this directive after a path to ensure
+  // that you don't log request bodies multiple times (simply because a matching test was made multiple times).
+  // TL;DR JUST PUT THIS THING AFTER YOUR FINAL PATH MATCHING
+  private def logRequestResponseHelper(
+      logIncomingRequest: (HttpRequest, RemoteAddress) => Future[Unit],
+      logResponse: HttpResponse => Future[Unit],
+  ): Directive0 =
+    extractRequest & extractClientIP tflatMap { case (request, remoteAddress) =>
+      mapRouteResultFuture { responseF =>
+        for {
+          _ <- logIncomingRequest(request, remoteAddress)
+          response <- responseF
+          _ <- response match {
+            case Complete(httpResponse) => logResponse(httpResponse)
+            case _ =>
+              Future.failed _ apply
+                new RuntimeException(
+                  """Logging the request & response should never happen on routes which get rejected.
+                    |Make sure to place the directive only at places where a match is guaranteed (e.g. after the path directive).""".stripMargin
+                )
+          }
+        } yield response
+      }
+    }
+
+  private def logJsonRequestAndResult(implicit
```

Comment:
I’m a bit confused, this seems like a fair amount of complexity when https://doc.akka.io/docs/akka-http/current/routing-dsl/directives/debugging-directives/logRequestResult.html seems like it should do the job. What problem does this solve that `logRequestResult` does not solve?

Reply:
I did not choose it, because it does not support using our contextualized logging (it seems the documentation is slightly outdated, the logging context parameter does not exist if I navigate the source in IntelliJ). However my implementation is pretty similar to the one from akka as I based on it :)

Reply:
Not sure contextualized logging matters all that much here given that this is purely for debugging but given that you already implemented it, we might as well keep it :+1: